### PR TITLE
Allow password as a callable when using Postgres backend

### PR DESCRIPTION
**Pull request by nwillems**
This change is intended for a case where you have changing passwords for the database, eg when using short-lived tokens.

Specifically, this allows one to use databases together with RDS IAM access, where a token is generated for new connections.

### CODE REVIEW

**Comment by nwillems** [comment]
I'm at a loss here - not quite sure how to get to those 100% coverage. Any help would be greatly appreciated :-)

**Comment by nwillems** [comment]
> 
> 
> I'm at a loss here - not quite sure how to get to those 100% coverage. Any help would be greatly appreciated :-)

Found out that it was coverage of the supplied function that was missing.

Typing is now also added.

Should be golden :-)

File: databases/backends/postgres.py [hunk 1]
```diff
@@ -67,7 +71,6 @@ async def connect(self) -> None:
             host=self._database_url.hostname,
             port=self._database_url.port,
             user=self._database_url.username,
-            password=self._database_url.password,
```
Comment (vmarkovtsev) [inline databases/backends/postgres.py:70]: I'd rather do it with fewer changes. We can get away with the following:
1. Revert everything.
2.
```python
kwargs = dict(
    host=self._database_url.hostname,
    port=self._database_url.port,
    user=self._database_url.username,
    password=self._database_url.password,
    database=self._database_url.database,
)
kwargs.update(self._get_connection_kwargs())
self._pool = await asyncpg.create_pool(**kwargs)
```
You can then pass `password` to `options / self._options`.

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]
Thanks for your contribution!

File: databases/backends/postgres.py (hunk 1 above)
Comment (nwillems) [reply databases/backends/postgres.py:70]: I agree. Like the updated code?

**Comment by nwillems** [comment]
@vmarkovtsev I've updated the code and it should be good - thanks for the review. Hope it fits :-)

**Comment by vmarkovtsev** [comment]
Great, thanks for the ping :+1:

**Review by vmarkovtsev — APPROVED** [review]
Glad that it worked!

**Comment by ugtar** [comment]
looks like the fix for this actually also fixes a regression that makes it impossible to connect to a postgres db over a unix socket (#257 ). The PR I wrote for that is nearly identical to this. I guess I can close that issue and the associated PR.

**Comment by vmarkovtsev** [comment]
Thank you @ugtar 👍

**Pull request merged**